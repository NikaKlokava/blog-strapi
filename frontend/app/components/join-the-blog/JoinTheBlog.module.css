.form_wrapper {
  position: relative;
  width: 80%;
  padding: 20px;
}

.form_container {
  width: 100%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  row-gap: 20px;
  align-items: center;
  border: 3px dotted #e0b8c6;
  border-radius: 5px;
  background-color: white;
}

.title {
  border-bottom: 3px dotted #E0B8C6;
}


.form_wrapper:before,
.form_wrapper:after {
  content: '';
  position: absolute;
  left: -2px;
  top: -2px;
  background: linear-gradient(45deg, #feebf2, #feebf2, #f3b2c9b5, #f3b2c9b5, #feebf2, #f3b2c9b5, #feebf2, #f3b2c9b5, #f3b2c9b5, #feebf2, #feebf2);
  background-size: 400%;
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  z-index: -1;
  animation: steam 30s linear infinite;

}

@keyframes steam {
  0% {
    background-position: 0 0;
    opacity: 0.5;
    border-radius: 15px;
  }

  50% {
    background-position: 400% 0;
    opacity: 0.5;
    border-radius: 15px;
  }

  100% {
    background-position: 0 0;
    opacity: 0.5;
    border-radius: 15px;
  }
}

.form_wrapper:after {
  filter: blur(5px);
}

.input_container {
  display: flex;
  column-gap: 20px;
}

.input_field {
  padding: 5px;
  border: 0.5px solid #000;
}

.submit_btn {
  min-width: 180px;
  padding: 5px;
  border: 2px solid #E0B8C6;
  border-radius: 20px;
  cursor: pointer;
  transition: 0.4s ease-in-out;
}

.submit_btn:hover {
  background-color: #e0b8c6;
}

.input_field:focus {
  outline: none;
}

.grid_style {
  grid-column: 1/3;
  justify-self: center;
}

@media screen and (max-width:1000px) {
  .form_wrapper {
    align-self: center;
  }
}

@media screen and (max-width:750px) {
  .form_wrapper {
    padding: 10px;
  }
}

@media screen and (max-width:550px) {
  .input_container {
    flex-direction: column;
    row-gap: 10px;
    width: 100%;
  }

  .form_container {
    row-gap: 10px;
  }

  .form_wrapper {
    width: 100%;
  }
}