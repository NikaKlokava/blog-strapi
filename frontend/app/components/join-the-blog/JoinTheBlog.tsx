"use client";

import styles from "./JoinTheBlog.module.css";
import { Formik, Form, Field } from "formik";
import { initialFormValue, validationSchema } from "@/app/utils/utils";
import classes from "classnames";

type Props = {
  extraClass?: string;
};

export const JoinTheBlog = ({ extraClass }: Props) => {
  return (
    <Formik
      initialValues={initialFormValue}
      validationSchema={validationSchema}
      onSubmit={(values) => {
        alert("Success");
        console.log(values);
      }}
    >
      {(props) => (
        <div
          className={classes(
            styles.form_wrapper,
            extraClass && styles[extraClass]
          )}
        >
          <Form
            className={styles.form_container}
            onSubmit={(e) => {
              e.preventDefault();
              props.handleSubmit();
            }}
          >
            <h1 className={styles.title}>Join the blog!</h1>
            <div className={styles.input_container}>
              <Field
                placeholder="your name"
                autoComplete="off"
                type="text"
                as="input"
                name={"name"}
                className={styles.input_field}
              />
              <Field
                placeholder="enter your email"
                autoComplete="off"
                type="email"
                as="input"
                name={"email"}
                className={styles.input_field}
              />
            </div>
            {props.errors.name && props.dirty && <div>{props.errors.name}</div>}
            <button className={styles.submit_btn} type="submit">
              Submit
            </button>
          </Form>
        </div>
      )}
    </Formik>
  );
};
