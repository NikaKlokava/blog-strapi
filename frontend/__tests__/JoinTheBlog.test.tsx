import { JoinTheBlog } from "@/app/components/join-the-blog/JoinTheBlog";
import { render } from "@testing-library/react";

describe("Test the JoinTheBlog component", () => {
  it("The JoinTheBlog component renders correctly", () => {
    jest.mock("../app/components/join-the-blog/JoinTheBlog");
    const joinTheBlogSnap = render(<JoinTheBlog />);
    expect(joinTheBlogSnap).toMatchSnapshot();
  });
});
