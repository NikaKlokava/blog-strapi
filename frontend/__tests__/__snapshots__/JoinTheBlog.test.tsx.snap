// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the JoinTheBlog component The JoinTheBlog component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form_wrapper"
      >
        <form
          action="#"
          class="form_container"
        >
          <h1
            class="title"
          >
            Join the blog!
          </h1>
          <div
            class="input_container"
          >
            <input
              autocomplete="off"
              class="input_field"
              name="name"
              placeholder="your name"
              type="text"
              value=""
            />
            <input
              autocomplete="off"
              class="input_field"
              name="email"
              placeholder="enter your email"
              type="email"
              value=""
            />
          </div>
          <button
            class="submit_btn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form_wrapper"
    >
      <form
        action="#"
        class="form_container"
      >
        <h1
          class="title"
        >
          Join the blog!
        </h1>
        <div
          class="input_container"
        >
          <input
            autocomplete="off"
            class="input_field"
            name="name"
            placeholder="your name"
            type="text"
            value=""
          />
          <input
            autocomplete="off"
            class="input_field"
            name="email"
            placeholder="enter your email"
            type="email"
            value=""
          />
        </div>
        <button
          class="submit_btn"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
